torch/csrc/jit/passes/quantization.cpp

```diff
@@ -752,6 +752,31 @@ bool isWeightOfConvOrLinear(Value* v) {
   return result;
 }
 
+// Go through the CallMethod graph to check if the value is Weight.
+c10::optional<bool> isWeight(Module& module, Value* v) {
+  if (isWeightOfConvOrLinear(v)) {
+    return true;
+  }
+  c10::optional<bool> result;
+  auto* self = v->owningGraph()->inputs()[0];
+  for (const Use& u : v->uses()) {
+    Node* n = u.user;
+    if (n->kind() == prim::CallMethod) {
+      auto m = getInvokedModule(module, n, self);
+      auto g = m.get_method(n->s(attr::name)).graph();
+      auto call_method_result = isWeight(m, g->inputs()[u.offset]);
+      if (result) {
+        // Check to make sure all the CallMethods in the graph produce the same output.
+        TORCH_CHECK(*call_method_result == *result,
+                      "Expected all CallMethods in the graph to produce same result.")
```

Comment:
can you have a better error message e.g. `"Expect the value either to be non-weight or weight:", v->debugName()`